Update all the things and remove some other stuff

grdryn opened the pull request:
Most notable changes:
- Update to latest dependency versions of everything available. Since we might end up getting `grunt@1+` as a peerDependency, we should have the 1+ of grunt-contrib-\* stuff, or we're gonna have a bad time.
- Remove `fh:plato` (or analysis or whatever it's called) since now that we're not using it, it just gives us dependency bloat
- Remove `fh:shrinkwrap`. This didn't really work well and we don't use it (I hope?)
- Use eslint instead of jshint (cherry-picked from @paolobueno in #12). This is probably the most breaking change, since depending projects will need to have .eslintrc.json files instead of .jshintrc files. This change is made because our [code style guide says we should be using it](https://github.com/fheng/help/blob/master/code-style.md). Let's just get it over with.
- Add 'porcelain' tasks for easier discovery/completion/helptext.

paolobueno committed:
Replace jshint stuff with eslint

paolobueno committed:
Fix linting errors

grdryn committed:
Update dependency versions

fh-automatron commented:
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:
- The code changes conform to the contributing & code style guides for this component (if they exist)
- A component owner should review your change (we've tagged these above)
- Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
- There should be an associated fhcap PR with this change - please provide a link to this
- If updating server component this PR should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
- Ensure JIRA is up to date before & after merge (e.g. deployment notes, tester notes, release notes, fix version)
- Any SonarQube issues should be addressed where applicable (use your own judgement)

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.

grdryn committed:
Use config.get instead of template.process

Motivation:
It seems like in grunt@1+, if a thing doesn't exist, then grunt will
fail while trying to read it with grunt.template.process.

Modification:
Use grunt.config.get everywhere instead.

Result:
We don't have to specify an empty array for fhignore and fhinclude in
all Gruntfile.js files that don't have anything else to put in them.

grdryn commented on package.json:
```diff
@@ -1,7 +1,7 @@
   "name": "grunt-fh-build",
   "description": "A plugin for development and build lifecycle of FeedHenry components",
-  "version": "0.5.0",
+  "version": "1.0.0",
```
Needs sonar file update

grdryn commented on package.json:
```diff
@@ -19,7 +19,7 @@
   },
   "devDependencies": {
-    "grunt": "~0.4.5"
+    "grunt": ">=0.4.0"
```
To match peerDependency that it asks for below.

grdryn committed:
Remove nsp since not exposed by any task:target

grdryn committed:
Remove underscore since lodash there also

grdryn committed:
Remove fh:plato and fh:shrinkwrap

Motivation:
fh:plato isn't used any longer, and fh:shrinkwrap should never have
existed.

Modification:
Remove these task targets and associated config and dependencies.

Result:
Fewer dependencies and things to worry about.

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
-          argsArray.forEach(function(arg) {
-            cmd += ' ' + arg;
-          });
+          var argsArray = Array.prototype.slice.call(arguments)
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
-        command: function() {
-          var cmd = './node_modules/.bin/istanbul report';
+        command: function () {
+          var cmd = './node_modules/.bin/istanbul report'
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
-          });
+          var argsArray = Array.prototype.slice.call(arguments)
+          argsArray.forEach(function (arg) {
+            cmd += ' ' + arg
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
-            cmd += ' ' + arg;
-          });
+          var argsArray = Array.prototype.slice.call(arguments)
+          argsArray.forEach(function (arg) {
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
+          })
 
-          return cmd;
+          return cmd
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
+  })
 
-  var checkPlaceholderFileExists = function() {
+  var checkPlaceholderFileExists = function () {
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -13,28 +13,26 @@
-var fs = require('fs');
-var _ld = require('lodash');
+var path = require('path')
+var findup = require('findup-sync')
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
-            var cmdArray = cmdsString.split(',');
-            cmd = cmdArray.join(' && ');
+          if (test_type && grunt.config.get(test_type)) {
+            var cmdsString = grunt.config.get(test_type)
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -13,28 +13,26 @@
-var _ = require('underscore');
-var fs = require('fs');
-var _ld = require('lodash');
+var path = require('path')
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -13,28 +13,26 @@
+var path = require('path')
+var findup = require('findup-sync')
+var _ = require('lodash')
+var fs = require('fs')
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
-        command: function(test_type) {
-          var cmd = '';
+        command: function (test_type) {
+          var cmd = ''
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -13,28 +13,26 @@
-var _ld = require('lodash');
+var path = require('path')
+var findup = require('findup-sync')
+var _ = require('lodash')
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -13,28 +13,26 @@
+var _ = require('lodash')
+var fs = require('fs')
 
+module.exports = function (grunt) {
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
+            cmd = cmdArray.join(' && ')
           } else {
-            grunt.warn("Invalid or missing parameter to " + this.nameArgs);
+            grunt.log.warn('Skipping', grunt.task.current.nameArgs,
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
-            cmd = cmdArray.join(' && ');
+          if (test_type && grunt.config.get(test_type)) {
+            var cmdsString = grunt.config.get(test_type)
+            var cmdArray = cmdsString.split(',')
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
+          if (test_type && grunt.config.get(test_type)) {
+            var cmdsString = grunt.config.get(test_type)
+            var cmdArray = cmdsString.split(',')
+            cmd = cmdArray.join(' && ')
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -13,28 +13,26 @@
+var fs = require('fs')
 
+module.exports = function (grunt) {
+  'use strict'
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -13,28 +13,26 @@
 
-module.exports = function(grunt) {
-  'use strict';
+  require('time-grunt')(grunt)
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
           }
 
-          return cmd;
+          return cmd
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -13,28 +13,26 @@
-  var dependencies = require('../package.json').dependencies;
-  var basepath = path.resolve(__dirname, '..');
+  var tasksDir
+  var dependencies = require('../package.json').dependencies
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -13,28 +13,26 @@
-  var tasksDir;
-  var dependencies = require('../package.json').dependencies;
-  var basepath = path.resolve(__dirname, '..');
+  var tasksDir
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -13,28 +13,26 @@
-  var basepath = path.resolve(__dirname, '..');
+  var tasksDir
+  var dependencies = require('../package.json').dependencies
+  var basepath = path.resolve(__dirname, '..')
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
+    var testType = 'unit_single'
     // Task filename parameter (used as template in runtime script)
-    grunt.config.set("unit_test_filename", testFile);
+    grunt.config.set('unit_test_filename', testFile)
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -13,28 +13,26 @@
+        {cwd: basepath})
       if (tasksDir) {
-        grunt.loadTasks(tasksDir);
+        grunt.loadTasks(tasksDir)
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -13,28 +13,26 @@
 
   for (var dependency in dependencies) {
     if (dependency.indexOf('grunt-') === 0) {
       tasksDir = findup(path.join('node_modules', dependency, 'tasks'),
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
+    grunt.config.set('unit_test_filename', testFile)
     for (var index = 1; index < args.length; index++) {
-      grunt.config.set("unit_test_param" + index, args[index]);
+      grunt.config.set('unit_test_param' + index, args[index])
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
+      grunt.config.set('unit_test_param' + index, args[index])
     }
-    var cmdArray = grunt.config.get(testType);
+    var cmdArray = grunt.config.get(testType)
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
     if (!cmdArray) {
-      grunt.log.errorlns("No local config for single tests. Please define " + testType + " property in config.");
-      return;
+      grunt.log.errorlns('No local config for single tests. Please define ' + testType + ' property in config.')
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
-      grunt.log.errorlns("No local config for single tests. Please define " + testType + " property in config.");
-      return;
+      grunt.log.errorlns('No local config for single tests. Please define ' + testType + ' property in config.')
+      return
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
-
-  });
-
+    grunt.task.run(['shell:fh-run-array:' + testType])
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
+    grunt.task.run(['shell:fh-run-array:' + testType])
+  }
 
+  grunt.registerTask('fh-generate-dockerised-config', 'Task to generate openshift config file', function () {
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -43,7 +41,7 @@ module.exports = function(grunt) {
    * its elements will be excluded from the tarball.
    */
-  var mergePatterns = function(bundle_deps) {
+  var mergePatterns = function (bundle_deps) {
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
 
+  grunt.registerTask('fh-generate-dockerised-config', 'Task to generate openshift config file', function () {
+    var conf = require(path.resolve('config/dev.json'))
+    var placeholders = require(path.resolve('config/ose-placeholders.js'))
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
+  }
 
+  grunt.registerTask('fh-generate-dockerised-config', 'Task to generate openshift config file', function () {
+    var conf = require(path.resolve('config/dev.json'))
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -43,7 +41,7 @@ module.exports = function(grunt) {
    */
-  var mergePatterns = function(bundle_deps) {
+  var mergePatterns = function (bundle_deps) {
     var patterns = [
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

fh-qube commented on tasks/fh-build.js:
```diff
@@ -243,174 +222,162 @@ module.exports = function(grunt) {
     try {
-      require(path.resolve('config/ose-placeholders.js'));
-      return true;
+      require(path.resolve('config/ose-placeholders.js'))
```
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

grdryn committed:
Add 'porcelain' task aliases for easier discovery

Motivation:
With the existing 'fh' multiTask, there isn't a good way to expose
available individual targets (subtasks), and what they can do. The
reason that we created them as targets of the 'fh' task initially,
rather than tasks in their own right, was just to namespace them from
other custom tasks that depending projects may have.

E.g. `fh:test` isn't going to clobber or get clobbered by any `test`
task that a project has defined in a Gruntfile.js or imported from a
plugin.

Modification:
New tasks have been added for many of the subtasks, and a short
description has been added for each of these.

Result:
If a developer has Grunt completion configured for their shell, they'll
be able to tab-complete to the task that they want.
Additionally, running `grunt --help` will show the description as help
text for each task, to make it clearer what each task does.

grdryn committed:
Remove CHANGELOG.md that we don't use any longer

grdryn commented on tasks/fh-build.js:
```diff
@@ -70,13 +69,15 @@ module.exports = function(grunt) {
     }
 
-    var fhignore = grunt.template.process('<%= fhignore %>');
+    var fhignore = grunt.config.get('fhignore');
```
In `grunt@1.0.0` if processing a template failed to resolve something (in this case, `fhignore` may not be defined in a `Gruntfile.js` of a project), then the command fails. Our implementation here depends on it ignoring that (we sanity check type below on line 74), and `grunt.config.get` seems fine with returning `undefined` as we expect if nothing is there.

grdryn commented on tasks/fh-build.js:
```diff
@@ -403,14 +382,43 @@ module.exports = function(grunt) {
       grunt.fail.warn('Unknown target provided to `grunt fh`');
     }
   });
+
```
We could probably replace the pre-existing `fh` multi-task with this rather than having both. I'm not sure if it's worth it right now though. I'd love to hear other thoughts on it!

wtrocki replied:
Great work! `fh-dist` looks way better than names containing `:` - this is used mostly when we need to pass some parameters to tasks.

grdryn replied:
Thanks for the feedback @wtrocki, I agree :)

I will probably remove the `fh-dist-no-bundle-deps` and `fh-dist-only-bundle-deps` before merging, because it turns out that that's not how the parameters are passed to them.

jasonmadigan replied:
+1, like ditching `:` in the task names

jasonmadigan commented:
@grdryn I'm a +1 on landing this. Did you have a plan to deliver updates to all components? The JSHint => ESLint changes will take a little time to land for each, but I'd guess we could take a few at a time. Nothing blocking us landing these changes individually - they don't (thankfully) have to land together..

grdryn commented:
@jasonmadigan Thanks for the feedback. There's no plan other than "do all the things as quickly as possible." :) I don't think it'll be too complicated, especially with `eslint --fix` or whatever. wdyt?

jasonmadigan commented:
@grdryn sounds good to me. With `--fix`, most of the mundane stuff is tackled. I had a look at NGUI for instance, and it might take a couple of hours to land the change.

paolobueno commented:
Glad to see the push for eslint being landed on fh-build. On this
transition I think it's a pretty easy choice to make, since jshint was
already failing for most modules (in the sense of having to use `-force`
for running subsequent tasks and the output being uselessly ignored). I'm
not very familiar with how the Jenkins jobs handled this so far, though.

So with eslint being easier to configure on which kinds of errors to fail
or warn, coupled with `--fix` mean all these small lint errors have an
actual chance of being fixed.

@grdryn https://github.com/grdryn sounds good to me. With --fix, most of
the mundane stuff is tackled. I had a look at NGUI for instance, and it
might take a couple of hours to land the change.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub
https://github.com/feedhenry/grunt-fh-build/pull/13#issuecomment-208253380

grdryn commented:
@jasonmadigan @paolobueno Yeah, a lot of the jobs currently don't even currently run jshint. If, when we're updating these it turns out that adding eslint will be straightforward, we can do it. If it's going to take a while, let's create separate issues for each one, so that they don't hold up the overall progress.

@paolobueno I'll send you a link to where our jenkins configs are, since that's a private repo somewhere else.

grdryn committed:
Remove tasks that don't work as expected

grdryn commented:
Going to merge this shortly then & release as 1.0.0, if there are no objections.

grdryn committed:
Set min node engine to 0.10.x

Motivation:
Most of the time, we'll get grunt@1.x.x from now on, which no longer
supports 0.8.x of NodeJS.

Modification:
Set min engine to 0.10.0

Result:
Probably just a warning when installing, if someone is using an earlier
version of NodeJS for some reason.

grdryn commented:
@lgriffin @jasonmadigan I'm considering adding one more change here: blacklisting `sonar-project.properties` from the built archives. Any thoughts on that?

paolobueno commented:
I can't really comment on the impact of that, but I also recommend that we look into `.npmignore` per module to remove other unneeded files and reduce the size of the published packages, if that's the intent.

jasonmadigan commented:
@grdryn https://docs.npmjs.com/misc/developers#keeping-files-out-of-your-package

jasonmadigan commented:
@grdryn beaten by @paolobueno!

jasonmadigan commented:
Btw, I'm down with merging & publishing this now. We can come up with some JIRAs for individual PITA components to `--fixup` ESLint on them if they're big tasks (`fh-supercore` & `fh-ngui` will probably take a few hours rather than minutes for example).

grdryn commented:
@jasonmadigan @paolobueno Up to now, we've been using this: https://github.com/feedhenry/grunt-fh-build/blob/master/tasks/fh-build.js#L47..L64

There are some other things in there than just "don't pack this"...it excludes the `package.json` for example, since we pack a slightly modified version of it in a lot of cases (modify `version` to include build number if there is one).

I'll just add it to that list for now, we can move more towards `.npmignore` in future, since that seems like a more standard way of doing things :)

grdryn committed:
Exclude sonar-project.properties from tar archive
